services/user-svc/internal/organization/organization.go

```diff
@@ -488,15 +518,29 @@ func GetOrganizationById(db *gorm.DB, id uuid.UUID) (*Organization, error) {
 }
 
 func GetInvitationById(db *gorm.DB, id uuid.UUID) (*Invitation, error) {
-	invite := Invitation{
+	invitation := Invitation{
 		ID: id,
 	}
 
-	if err := db.First(&invite).Error; err != nil {
+	if err := db.First(&invitation).Error; err != nil {
 		return nil, err
 	}
 
-	return &invite, nil
+	return &invitation, nil
+}
+
+func GetInvitationByIdAndEmail(db *gorm.DB, email string, id uuid.UUID) (*Invitation, error) {
+
+	var invitation Invitation
+	if err := db.Where("id = ? AND email = ?", id, email).First(&invitation).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
```

Comment:
```suggestion
		return nil, err
```
Do the error handling inside the "controller", one layer up?